features/dd-sdk-android-session-replay/src/test/kotlin/com/datadog/android/sessionreplay/internal/recorder/base64/BitmapPoolTest.kt

```diff
@@ -0,0 +1,216 @@
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.sessionreplay.internal.recorder.base64
+
+import android.graphics.Bitmap
+import android.util.LruCache
+import com.datadog.android.sessionreplay.forge.ForgeConfigurator
+import fr.xgouchet.elmyr.Forge
+import fr.xgouchet.elmyr.junit5.ForgeConfiguration
+import fr.xgouchet.elmyr.junit5.ForgeExtension
+import org.assertj.core.api.Assertions
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.junit.jupiter.api.extension.Extensions
+import org.mockito.Mock
+import org.mockito.junit.jupiter.MockitoExtension
+import org.mockito.junit.jupiter.MockitoSettings
+import org.mockito.kotlin.verify
+import org.mockito.kotlin.whenever
+import org.mockito.quality.Strictness
+
+@Extensions(
+    ExtendWith(MockitoExtension::class),
+    ExtendWith(ForgeExtension::class)
+)
+@MockitoSettings(strictness = Strictness.LENIENT)
+@ForgeConfiguration(ForgeConfigurator::class)
+internal class BitmapPoolTest {
+    @Mock
+    lateinit var mockLruCache: LruCache<String, Bitmap>
+
+    @Mock
+    lateinit var mockBitmap: Bitmap
+
+    @Mock
+    lateinit var mockSecondBitmap: Bitmap
+
+    @Mock
+    lateinit var mockConfig: Bitmap.Config
+
+    private val testedCache = BitmapPool
+
+    @BeforeEach
+    fun setup(forge: Forge) {
+        testedCache.setBackingCache(mockLruCache)
+
+        val width = forge.anInt(1, 200)
+        val height = forge.anInt(1, 200)
+        val allocSize = forge.anInt(100, 10000)
+
+        whenever(mockBitmap.config).thenReturn(mockConfig)
+        whenever(mockSecondBitmap.config).thenReturn(mockConfig)
+        whenever(mockBitmap.width).thenReturn(width)
+        whenever(mockBitmap.height).thenReturn(height)
+        whenever(mockBitmap.allocationByteCount).thenReturn(allocSize)
+        whenever(mockSecondBitmap.width).thenReturn(width)
+        whenever(mockSecondBitmap.height).thenReturn(height)
+        whenever(mockSecondBitmap.allocationByteCount).thenReturn(allocSize)
+    }
+
+    @Test
+    fun `M return null W get() { cache has no bitmaps of the right size }`() {
+        // Given
+        val width = mockBitmap.width
+        val height = mockBitmap.height
+        val config = mockConfig
+
+        // When
+        val bitmap = testedCache.getBitmapByProperties(width, height, config)
+
+        // Then
+        assertThat(bitmap).isNull()
+    }
+
+    @Test
+    fun `M return a bitmap W get() { free bitmap in cache }`() {
+        // Given
+        val width = mockBitmap.width
+        val height = mockBitmap.height
+        val config = mockConfig
+        val key = "$width-$height-$config"
+        whenever(mockLruCache.get(key)).thenReturn(mockBitmap)
```

Comment:
why mocking the cache here and not using the logic inside by leveraging the `put` method ? This way you can test also the add to cache logic into this method.